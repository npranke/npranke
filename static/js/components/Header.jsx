import { NavLink } from 'react-router-dom-v5-compat'
import PropTypes from 'prop-types'
import React from 'react'

import GitHub from '@img/icon-github.png'
import Home from '@img/icon-home.png'
import LinkedIn from '@img/icon-linkedin.png'
import Workbook from '@img/icon-workbook.png'

import { getSendEventHandler } from '@utils'

function Header(props) {
    const homeLink = props.isPageNotFound
        ? (
            <a
                onClick={ getSendEventHandler(
                    'header',
                    'navigate',
                    'home',
                ) }
                href="/home"
                target="_self"
                rel="noreferrer"
                title="Home"
                aria-label="Home"
            >
                <img src={ Home } className="icon icon-home" alt="Home icon" />
            </a>
        ) : (
            <NavLink
                end
                onClick={ getSendEventHandler(
                    'header',
                    'navigate',
                    'home',
                ) }
                to="/home"
                target="_self"
                rel="noreferrer"
                className="navlink navlink-home"
                title="Home"
                aria-label="Home"
            >
                <img src={ Home } className="icon icon-home" alt="Home icon" />
            </NavLink>
        )

    const workbookLink = props.isPageNotFound
        ? (
            <a
                onClick={ getSendEventHandler(
                    'header',
                    'navigate',
                    'workbook',
                ) }
                href="/workbook"
                target="_self"
                rel="noreferrer"
                title="Workbook"
                aria-label="Workbook"
            >
                <img
                    src={ Workbook }
                    className="icon icon-workbook"
                    alt="Workbook icon"
                />
            </a>
        ) : (
            <NavLink
                onClick={ getSendEventHandler(
                    'header',
                    'navigate',
                    'workbook',
                ) }
                to="/workbook"
                target="_self"
                rel="noreferrer"
                className="navlink navlink-workbook"
                title="Workbook"
                aria-label="Workbook"
            >
                <img
                    src={ Workbook }
                    className="icon icon-workbook"
                    alt="Workbook icon"
                />
            </NavLink>
        )

    return (
        <nav className="header">
            <div className="left-header">
                <div className="icon-container icon-container-home">
                    { homeLink }
                </div>
            </div>
            <div className="right-header">
                <div className="icon-container icon-container-workbook">
                    { workbookLink }
                </div>
                <div className="icon-container icon-container-github">
                    <a
                        onClick={ getSendEventHandler(
                            'header',
                            'navigate',
                            'github',
                        ) }
                        href="https://github.com/npranke"
                        target="_blank"
                        rel="noreferrer noopener external"
                        title="GitHub"
                        aria-label="GitHub"
                    >
                        <img
                            src={ GitHub }
                            className="icon icon-github"
                            alt="GitHub icon"
                        />
                    </a>
                </div>
                <div className="icon-container icon-container-linkedin">
                    <a
                        onClick={ getSendEventHandler(
                            'header',
                            'navigate',
                            'linkedin',
                        ) }
                        href="https://www.linkedin.com/in/npranke"
                        target="_blank"
                        rel="noreferrer noopener external"
                        title="LinkedIn"
                        aria-label="LinkedIn"
                    >
                        <img
                            src={ LinkedIn }
                            className="icon icon-linkedin"
                            alt="LinkedIn icon"
                        />
                    </a>
                </div>
            </div>
        </nav>
    )
}

Header.defaultProps = {
    isPageNotFound: false,
}

Header.propTypes = {
    isPageNotFound: PropTypes.bool,
}

export default Header
