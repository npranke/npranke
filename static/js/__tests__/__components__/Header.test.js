import Adapter from 'enzyme-adapter-react-16'
import Enzyme, { shallow } from 'enzyme'
import React from 'react'

import Header from '../../components/Header'

Enzyme.configure({ adapter: new Adapter() })

describe('Header', () => {
    test('has link to home when pagenotfound', () => {
        const header = shallow(<Header isPageNotFound />)

        expect(
            header.find('.icon-container-home a').props().href,
        ).toEqual('/home')
    })

    test('has link to workbook when pagenotfound', () => {
        const header = shallow(<Header isPageNotFound />)

        expect(
            header.find('.icon-container-workbook a').props().href,
        ).toEqual('/workbook')
    })

    test('has link to home when not pagenotfound', () => {
        const header = shallow(<Header />)

        expect(
            header.find('.icon-container-home NavLink').props().to,
        ).toEqual('/home')
    })

    test('has link to workbook when not pagenotfound', () => {
        const header = shallow(<Header />)

        expect(
            header.find('.icon-container-workbook NavLink').props().to,
        ).toEqual('/workbook')
    })

    test('has link to github profile', () => {
        const header = shallow(<Header />)

        expect(
            header.find('.icon-container-github a').props().href,
        ).toEqual('https://github.com/npranke')
    })

    test('has link to linkedin profile', () => {
        const header = shallow(<Header />)

        expect(
            header.find('.icon-container-linkedin a').props().href,
        ).toEqual('https://www.linkedin.com/in/npranke')
    })

    test('has alt text for home icon', () => {
        const header = shallow(<Header />)

        expect(
            header.find('.icon-home').props().alt,
        ).toEqual('Home icon')
    })

    test('has alt text for workbook icon', () => {
        const header = shallow(<Header />)

        expect(
            header.find('.icon-workbook').props().alt,
        ).toEqual('Workbook icon')
    })

    test('has alt text for github icon', () => {
        const header = shallow(<Header />)

        expect(
            header.find('.icon-github').props().alt,
        ).toEqual('GitHub icon')
    })

    test('has alt text for linkedin icon', () => {
        const header = shallow(<Header />)

        expect(
            header.find('.icon-linkedin').props().alt,
        ).toEqual('LinkedIn icon')
    })
})

describe('Header snapshot', () => {
    test('matches snapshot when pagenotfound', () => {
        const header = shallow(<Header isPageNotFound />)

        expect(header).toMatchSnapshot()
    })

    test('matches snapshot when not pagenotfound', () => {
        const header = shallow(<Header />)

        expect(header).toMatchSnapshot()
    })
})
