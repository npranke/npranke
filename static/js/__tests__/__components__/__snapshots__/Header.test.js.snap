// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header snapshot matches snapshot 1`] = `
<DocumentFragment>
  <nav
    class="header"
  >
    <div
      class="left-header"
    >
      <div
        class="icon-container icon-container-home"
      >
        <a
          aria-label="Home"
          class="navlink navlink-home"
          href="/home"
          rel="noreferrer"
          target="_self"
          title="Home"
        >
          <img
            alt="Home icon"
            class="icon icon-home"
            src="image-mock"
          />
        </a>
      </div>
      <div
        class="icon-container icon-container-workbook"
      >
        <a
          aria-label="Workbook"
          class="navlink navlink-workbook"
          href="/workbook"
          rel="noreferrer"
          target="_self"
          title="Workbook"
        >
          <img
            alt="Workbook icon"
            class="icon icon-workbook"
            src="image-mock"
          />
        </a>
      </div>
    </div>
    <div
      class="right-header"
    >
      <div
        class="icon-container icon-container-github"
      >
        <a
          aria-label="GitHub"
          href="https://github.com/npranke"
          rel="noreferrer noopener external"
          target="_blank"
          title="GitHub"
        >
          <img
            alt="GitHub icon"
            class="icon icon-github"
            src="image-mock"
          />
        </a>
      </div>
      <div
        class="icon-container icon-container-linkedin"
      >
        <a
          aria-label="LinkedIn"
          href="https://www.linkedin.com/in/npranke"
          rel="noreferrer noopener external"
          target="_blank"
          title="LinkedIn"
        >
          <img
            alt="LinkedIn icon"
            class="icon icon-linkedin"
            src="image-mock"
          />
        </a>
      </div>
    </div>
  </nav>
</DocumentFragment>
`;
